pkg/cmd/chat/push/push.go

```diff
@@ -0,0 +1,155 @@
+			$ stream-cli chat upsert-pushprovider --properties "{'type': 'huawei', 'name': 'staging', 'huawei_app_id': 'id', 'huawei_app_secret': 'secret'}"
+
+			# Setting up a Xiaomi push provider
+			$ stream-cli chat upsert-pushprovider --properties "{'type': 'xiaomi', 'name': 'staging', 'xiaomi_package_name': 'name', 'xiaomi_app_secret': 'secret'}"
+		`),
+		RunE: func(cmd *cobra.Command, args []string) error {
+			c, err := config.GetConfig(cmd).GetClient(cmd)
+			if err != nil {
+				return err
+			}
+
+			prop, _ := cmd.Flags().GetString("properties")
+
+			var p stream_chat.PushProvider
+			err = json.Unmarshal([]byte(prop), &p)
+			if err != nil {
+				return err
+			}
+
+			_, err = c.UpsertPushProvider(cmd.Context(), &p)
+			if err != nil {
+				return err
+			}
+
+			cmd.Println("Successfully updated push provider.")
+			return nil
+		},
+	}
+
+	fl := cmd.Flags()
+	fl.StringP("properties", "p", "", "[required] Raw json properties to send to the backend")
+	_ = cmd.MarkFlagRequired("properties")
+
+	return cmd
+}
+
+func testCmd() *cobra.Command {
+	cmd := &cobra.Command{
+		Use: "test-push --message-id [string]" +
+			" --apn-template [string]" +
+			" --firebase-template [string]" +
+			" --firebase-data-template [string]" +
+			" --skip-devices [true|false]" +
+			" --push-provider-name [string]" +
+			" --push-provider-type [string]" +
+			" --user-id [string]" +
+			" --output-format [json|tree]",
+		Short: "Test push notifications",
+		Example: heredoc.Doc(`
+			# A test push notification for a certain message id
+			$ stream-cli chat test-push --message-id msgid --user-id id --skip-devices true
+		`),
+		RunE: func(cmd *cobra.Command, args []string) error {
+			c, err := config.GetConfig(cmd).GetClient(cmd)
+			if err != nil {
+				return err
+			}
+
+			msgID, _ := cmd.Flags().GetString("message-id")
+			apnTemplate, _ := cmd.Flags().GetString("apn-template")
+			firebaseTemplate, _ := cmd.Flags().GetString("firebase-template")
+			firebaseDataTemplate, _ := cmd.Flags().GetString("firebase-data-template")
+			skipDevices, _ := cmd.Flags().GetBool("skip-devices")
+			pushProviderName, _ := cmd.Flags().GetString("push-provider-name")
+			pushProviderType, _ := cmd.Flags().GetString("push-provider-type")
+			userID, _ := cmd.Flags().GetString("user-id")
+
+			p := &stream_chat.CheckPushRequest{
+				MessageID:            msgID,
+				ApnTemplate:          apnTemplate,
+				FirebaseTemplate:     firebaseTemplate,
+				FirebaseDataTemplate: firebaseDataTemplate,
+				SkipDevices:          &skipDevices,
+				PushProviderName:     pushProviderName,
+				PushProviderType:     pushProviderType,
+				UserID:               userID,
+			}
+
+			resp, err := c.CheckPush(cmd.Context(), p)
+			if err != nil {
+				return err
+			}
+
+			return utils.PrintObject(cmd, resp)
+		},
+	}
+
+	fl := cmd.Flags()
+	fl.String("message-id", "", "[optional] Message id to test")
+	fl.String("apn-template", "", "[optional] APN template to send to the backend")
```

Comment:
apn-template, firebase-template, firebase-data-template are considered as legacy.
Let's not support them here.

Reply:
We can add them later if we need to

Reply:
yeah, I'll update it once I can merge my SDK pr. thanks 👍

Reply:
this is good to update now

Reply:
working on it today

Reply:
removed those 3